### ISSUE TITLE
Show functions for all the datatypes

### ISSUE THREAD

**Issue by gallais** [agda/agda-stdlib#431]
Currently we only have `Data.(Bool/Nat).Show`.

We should probably use something like [`showsPrec`](https://hackage.haskell.org/package/base-4.11.1.0/docs/Prelude.html#v:showsPrec) to avoid introducing too many parentheses.

**Comment by mechvel** [comment]
I guess now that you also meant a version of the Show class.

**Comment by silas-hw** [comment]
Is this something that still needs doing?

**Comment by JacquesCarette** [comment]
Yes - but incrementally. There is probably no point in being thorough/systematic about this (yet). But certainly more would be good, such as for `Data.Integer` and `Data.Rational`. And likely for `Maybe` and `List` as well.

**Comment by Taneb** [comment]
Aside: we already have added `Data.integer.Show` and `Data.Rational.Show`. But for container structures like `Maybe` and `List` it would be very nice

**Comment by silas-hw** [comment]
Containers would really need a Show class to exist (e.g. to have `showMaybe : {{ Show A} } -> Maybe A -> String`), i.e. #569, which seems a bigger task. Considering there's been a lot of discussion on the best-fit design, I'm unsure I'm suited to start up a PR for that.

**Comment by JacquesCarette** [comment]
Agree that containers are too big a chunk at the moment. Might be worth looking around to see if there are non-container data that don't have a `Show` yet. And simply move on to something else if there is nothing obvious.
